src/main/kotlin/com/github/treepat/expression/operators/TreepatFunction.kt

```diff
@@ -1,22 +1,5 @@
 package com.github.treepat.expression.operators
 
 fun treepatFunction(expression: VisitorFunction): VisitorFunction = { targetTreeNode ->
-    var currentNode = targetTreeNode
-    val answers = mutableListOf<VisitorFunctionResponse>()
-    while (currentNode != null) {
-        val response = expression.invoke(currentNode)
-        answers.add(
-            VisitorFunctionResponse(
-                response.responses.filter { it.matches.isNotEmpty() },
-                response.hasMatch
-            )
-        )
-        currentNode = currentNode.nextLeftmostPreorderNode()
-    }
-    val allHasMatches = answers.filter { it.hasMatch }.filter { it.responses.isNotEmpty() }.map { it.responses.last() }
-    if (allHasMatches.isNotEmpty()) {
-        VisitorFunctionResponse(allHasMatches, true)
-    } else {
-        VisitorFunctionResponseFactory.createResponseWithZeroMatches(targetTreeNode)
-    }
+    expression.invoke(targetTreeNode)
```

Comment:
As we didn't need TreeFunction anymore I just stripped it of all its functionality.

Reply:
I think we can delete this class then. :)

Reply:
Sure, alreadey getting rid of it